tests/test_job_management.py

```diff
@@ -1,95 +1,253 @@
 import requests
 import json
 import os
-import pytest
+import time
+import jwt
+import logging
+
+
+logger = logging.getLogger(__name__)
+# symmetric shared secret for the decoding of the token
+secret_key = 'secretkey_test'
+
+default_exp_time = int(time.time()) + 5000
+default_token_payload = dict(
+    sub="mtm@mtmco.net",
+    name="mmeharga",
+    roles="general",
+    exp=default_exp_time,
+    tem=0,
+    mstout=True,
+    mssub=True
+)
+
 
 def test_callback_without_prior_run_analysis(dispatcher_live_fixture):
     server = dispatcher_live_fixture
     print("constructed server:", server)
 
     c = requests.get(server + "/call_back",
-                   params={
-                       'job_id': 'test-job-id',
-                       'instrument_name': 'test-instrument_name',
-                   },
-                )
+                     params={
+                         'job_id': 'test-job-id',
+                         'instrument_name': 'test-instrument_name',
+                     })
 
-    print(c.text)    
+    print(c.text)
 
     assert c.status_code == 200
 
 
-def test_callback_after_run_analysis(dispatcher_live_fixture):
-    #TODO: for now, this is not very different from no-prior-run_analysis. This will improve
+def test_email_callback_after_run_analysis(dispatcher_live_fixture, dispatcher_local_mail_server):
+    # TODO: for now, this is not very different from no-prior-run_analysis. This will improve
 
     server = dispatcher_live_fixture
     print("constructed server:", server)
 
+    # let's generate a valid token with high threshold
+    token_payload = {
+        **default_token_payload,
+        "tem": 0,
+    }
+    encoded_token = jwt.encode(token_payload, secret_key, algorithm='HS256')
+    # set the time the request was initiated
+    time_request = time.time()
+    # this should return status submitted, so email sent
     c = requests.get(server + "/run_analysis",
                      params=dict(
-                        query_status="new",
-                        query_type="Real",
-                        instrument="empty-async",
-                        product_type="dummy",                        
-                    ))
+                         query_status="new",
+                         query_type="Real",
+                         instrument="empty-async",
+                         product_type="dummy",
+                         token=encoded_token
+                     ))
 
     print("response from run_analysis:", json.dumps(c.json(), indent=4))
 
     session_id = c.json()['session_id']
     job_id = c.json()['job_monitor']['job_id']
 
-    #TODO ensure it is submitted
-
     job_monitor_json_fn = f'scratch_sid_{session_id}_jid_{job_id}/job_monitor.json'
+    # the aliased version might have been created
```

Comment:
when I tested I assumed the area is empty, and alias is not created. It is true that it's better to ensure that, or workaround like you did.
I wonder if it is better to enforce this property, e.g. by setting some random parameter.
That way both path: aliased and not, would be tested clearly separately.
But I would not want this request to request extra work. Maybe lets add a TODO? What do you think?

Reply:
One question: the reason why the aliased folder is created is only because there was already a folder created with the same session_id and job_id?

Reply:
only the same job_id is required. The purpose is that if there is already a running job, the user can re-attach to it. That URL we insert in the email will lead the user to the same request.
But session ID may change, if new browser window is started.
By the way, we should prevent aliasing between different authorized users:

https://github.com/oda-hub/dispatcher-app/issues/59

Reply:
Ok, I will add as a TODO, it is not strictly related to this PR